### PR TITLE
Update response-compression.md

### LINKED ISSUE #22479: [ResponseCompressionDefaults] Add `application/wasm` to default MIME types list
Source commit: https://github.com/dotnet/aspnetcore/commit/2d3ad1cb0f9872da40c6d143ce4554563971ae04
Doc: https://docs.microsoft.com/en-us/aspnet/core/performance/response-compression#mime-types

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9795b92f-db24-8f1f-88c0-c36666d3526f
* Version Independent ID: 2d56d9b6-3d7b-0204-6c08-8e584fccecc2
* Content: [Response compression in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/response-compression?view=aspnetcore-5.0)
* Content Source: [aspnetcore/performance/response-compression.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/performance/response-compression.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by mmarinchenko**
Fixes #22479

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@mmarinchenko  thanks for the PR. Unfortunately creating a new moniker range is impossible for us to verify and review. I created a new 7.0 moniker range. Unfortunately GitHub is not very good and showing just the changes you made.

**Comment by mmarinchenko** [comment]
@Rick-Anderson thanks!👍

I've resolved conflicts so the proposed changes can now be reviewed.

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

File: aspnetcore/performance/response-compression.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ The headers involved in requesting, sending, caching, and receiving compressed c
...
-| [`Content-Type`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Type)     | Specifies the [MIME type](https://developer.mozilla.org/docs/Web/HTTP/Basics_of_HTTP/MIME_types) of the content. Every response should specify its `Content-Type`. The response compression middleware checks this value to determine if the response should be compressed. The response compression middleware specifies a set of [default MIME types](https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/ResponseCompression/src/ResponseCompressionDefaults.cs#L14-L30) that it can encode, and they can bed replaced or added. |
+| [`Content-Type`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Content-Type)     | Specifies the [MIME type](https://developer.mozilla.org/docs/Web/HTTP/Basics_of_HTTP/MIME_types) of the content. Every response should specify its `Content-Type`. The response compression middleware checks this value to determine if the response should be compressed. The response compression middleware specifies a set of [default MIME types](#mime-types) that it can encode, and they can bed replaced or added. |
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md:72]: The MIME types section contains exactly the same link to the source code. If this link needs to be changed in the future, one will need to find out that there are two of them in the document, and change both.

It is much more convenient to have one external link and refer to the corresponding section in the rest of the document. If it makes no sense to refer to a section inside a document, why do we need this section at all?

File: aspnetcore/performance/response-compression.md [hunk 2]
```diff
@@ -259,7 +261,7 @@ The headers involved in requesting, sending, caching, and receiving compressed c
...
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md:264]: Same here - https://github.com/dotnet/AspNetCore.Docs/pull/26173#discussion_r903271949

File: aspnetcore/performance/response-compression.md [hunk 3]
```diff
@@ -336,28 +338,29 @@ With the preceding code, the response body isn't compressed by the sample. Howev
 
 ## MIME types
 
-The response compression middleware specifies a [default set of MIME types for compression](https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/ResponseCompression/src/ResponseCompressionDefaults.cs#L14-L30):
+The response compression middleware specifies a [default set of MIME types for compression](https://github.com/dotnet/aspnetcore/blob/v6.0.0/src/Middleware/ResponseCompression/src/ResponseCompressionDefaults.cs#L19-L31):
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md:341]: I'm OK with that, but why duplicate (well, almost, but still) a link in two sentences?🙂

I think the first one should be eliminated then.

**Comment by mmarinchenko** [comment]
> I'll probably swap 6 and 7 but my questions is why do we even need a 7 branch of we point at the source code?

> What in the PR requires a 7.0 moniker section/version?

Without explicit MIME types list - nothing)

I removed explicit list and 7.0 moniker.

File: aspnetcore/performance/response-compression.md [hunk 4]
```diff
@@ -336,28 +338,29 @@ With the preceding code, the response body isn't compressed by the sample. Howev
...
+When compressing responses based on the [`Accept-Encoding` request header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding), there are can be uncompressed and multiple compressed versions of the response. In order to instruct client and proxy caches that multiple versions exist and should be stored, the `Vary` header is added with an `Accept-Encoding` value. The response middleware [adds the `Vary` header](https://github.com/dotnet/aspnetcore/blob/v6.0.0/src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs#L204-L247) automatically when the response is compressed.
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md]: Done as part of 7.0 moniker removal.

File: aspnetcore/performance/response-compression.md [hunk 5]
```diff
@@ -381,7 +384,7 @@ The [sample app](https://responsecompression.azurewebsites.net/) depolyed to Azu
 ## Additional resources
 
 * [View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/response-compression/samples/6.x) ([how to download](xref:index#how-to-download-a-sample))
-* [Response compression middleware source](https://github.com/dotnet/aspnetcore/tree/main/src/Middleware/ResponseCompression/src)
+* [Response compression middleware source](https://github.com/dotnet/aspnetcore/tree/v6.0.0/src/Middleware/ResponseCompression/src)
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md]: Done.

File: aspnetcore/performance/response-compression.md [hunk 6]
```diff
@@ -394,7 +397,7 @@ The [sample app](https://responsecompression.azurewebsites.net/) depolyed to Azu
 
 Network bandwidth is a limited resource. Reducing the size of the response usually increases the responsiveness of an app, often dramatically. One way to reduce payload sizes is to compress an app's responses.
 
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/response-compression/samples) ([how to download](xref:index#how-to-download-a-sample))
+[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/response-compression/samples/3.x) ([how to download](xref:index#how-to-download-a-sample))
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md]: Done.

File: aspnetcore/performance/response-compression.md [hunk 7]
```diff
@@ -445,15 +448,11 @@ The headers involved in requesting, sending, caching, and receiving compressed c
 | `Content-Type`     | Specifies the MIME type of the content. Every response should specify its `Content-Type`. The middleware checks this value to determine if the response should be compressed. The middleware specifies a set of [default MIME types](#mime-types) that it can encode, but you can replace or add MIME types. |
 | `Vary`             | When sent by the server with a value of `Accept-Encoding` to clients and proxies, the `Vary` header indicates to the client or proxy that it should cache (vary) responses based on the value of the `Accept-Encoding` header of the request. The result of returning content with the `Vary: Accept-Encoding` header is that both compressed and uncompressed responses are cached separately. |
 
-Explore the features of the Response Compression Middleware with the [sample app](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/response-compression/samples). The sample illustrates:
+Explore the features of the Response Compression Middleware with the [sample app](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/performance/response-compression/samples/3.x). The sample illustrates:
```
[the remark this answers is not in the document]
Comment (mmarinchenko) [reply aspnetcore/performance/response-compression.md]: Done.

**Comment by mmarinchenko** [comment]
> * The new 7.0 content is at the **top** of the article's markdown, so it didn't get your, Rick, workaround for broken links until we get to November.

Just for your information, I originally added the 7.0 content to the top because I found the same thing in the [Kestrel Endpoints](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/servers/kestrel/endpoints.md) document.

**Review by Rick-Anderson — APPROVED** [review]

**Pull request merged**